src/main/java/com/iexec/worker/feign/LoginService.java

```diff
@@ -37,6 +38,7 @@ public class LoginService {
 
     private final CredentialsService credentialsService;
     private final CoreClient coreClient;
+    private final Semaphore lock = new Semaphore(1);
```

Comment:
Shouldn't we use a `ReentrantLock` instead? It seems to be more suited to our case: only the thread that acquired the lock should be able to release it.

Reply:
I don't know, I don't like the reentrancy part as well which we want to avoid here.

Reply:
done in ae6b89d